src/Autofac/Core/Resolving/ResolveOperationBase.cs

```diff
@@ -83,48 +73,43 @@ protected ResolveOperationBase(ISharingLifetimeScope mostNestedLifetimeScope, IR
         }
 
         /// <summary>
-        /// Gets the active resolve request.
+        /// Gets or sets the active resolve request.
         /// </summary>
-        public ResolveRequestContextBase? ActiveRequestContext { get; private set; }
+        private ResolveRequestContextBase? ActiveRequestContext { get; set; }
 
         /// <summary>
-        /// Gets the current lifetime scope of the operation; based on the most recently executed request.
+        /// Gets or sets the current lifetime scope of the operation; based on the most recently executed request.
         /// </summary>
-        public ISharingLifetimeScope CurrentScope { get; private set; }
```

Comment:
The `CurrentScope` of the `ResolveOperation` is technically different from the `ActivationScope` of a `ResolveRequestContext`, because it represents the scope at the **start** of the Resolve, rather than one selected by the `ScopeSelectionMiddleware` during the resolve.